tests/vector_field_nets_test.py

```diff
@@ -0,0 +1,217 @@
+# This file is part of sbi, a toolkit for simulation-based inference. sbi is licensed
+# under the Apache License Version 2.0, see <https://www.apache.org/licenses/>
+
+from __future__ import annotations
+
+from typing import Tuple
+
+import pytest
+import torch
+import torch.nn as nn
+
+from sbi.neural_nets.embedding_nets import CNNEmbedding
+from sbi.neural_nets.net_builders.vector_field_nets import (
+    VectorFieldMLP,
+    VectorFieldTransformer,
+    build_adamlp_network,
+    build_standard_mlp_network,
+    build_transformer_network,
+)
+
+
+@pytest.mark.parametrize("input_event_shape", ((1,), (4,)))
+@pytest.mark.parametrize("condition_event_shape", ((1,), (7,)))
+@pytest.mark.parametrize("batch_dim", (1, 10))
+@pytest.mark.parametrize("net_type", ["mlp", "ada_mlp", "transformer"])
+@pytest.mark.parametrize("time_emb_type", ["sinusoidal", "random_fourier"])
+@pytest.mark.parametrize("activation", [nn.GELU, nn.ReLU, nn.SiLU])
+def test_network_shapes(
+    input_event_shape,
+    condition_event_shape,
+    batch_dim,
+    net_type,
+    time_emb_type,
+    activation,
+):
+    """Test whether vector field networks produce correct output shapes."""
+    network, inputs, conditions = _build_vector_field_components(
+        net_type,
+        input_event_shape,
+        condition_event_shape,
+        batch_dim,
+        time_emb_type=time_emb_type,
+        activation=activation,
+    )
+
+    # Create time parameter
+    t = torch.rand((batch_dim,))
+
+    # Test forward pass
+    outputs = network(inputs, conditions, t)
+    assert outputs.shape == inputs.shape, "Output shape should match input shape"
+
+
+@pytest.mark.parametrize("hidden_features", [50, 100, [50, 100]])
+@pytest.mark.parametrize("num_layers", [2, 5])
+@pytest.mark.parametrize("layer_norm", [True, False])
+@pytest.mark.parametrize("skip_connections", [True, False])
+@pytest.mark.parametrize("time_emb_type", ["sinusoidal", "random_fourier"])
+def test_mlp_network_parameters(
+    hidden_features, num_layers, layer_norm, skip_connections, time_emb_type
+):
+    """Test whether MLP vector field networks can be built with different parameters."""
+    batch_x = torch.randn(100, 5)
+    batch_y = torch.randn(100, 3)
+
+    network = build_standard_mlp_network(
+        batch_x=batch_x,
+        batch_y=batch_y,
+        hidden_features=hidden_features,
+        num_layers=num_layers,
+        time_embedding_dim=32,
+        layer_norm=layer_norm,
+        skip_connections=skip_connections,
+        time_emb_type=time_emb_type,
+    )
+
+    # Verify it's the correct type
+    assert isinstance(network, VectorFieldMLP), "Should be a VectorFieldMLP instance"
```

Comment:
Not really sure what this line tests for, as we specificalyl build with standard_mlp_network. Did we mean to use `build_vf_estimator` and test whether the net is correctly specified?